src/witan/models/dem/ccm/components_functions.clj

```diff
@@ -97,7 +97,30 @@
                                   (wds/safe-divide [(- national-trend last-year-nt) last-year-nt]))))
       (ds/select-columns [:sex :age :year :prop-diff])))
 
-(defn apply-national-trend
+
+;;recursive instead of loop-recur
+(defn apply-national-trend-to-data
```

Comment:
nitpick: I think this would go into a reduce again ;)  lmk if I'm wrong.  It's just that recursive functions, while working, end up being a little harder to reason about.